substrate/client/consensus/beefy/src/round.rs

```diff
@@ -126,6 +163,14 @@ where
 		self.validator_set.validators()
 	}
 
+	/// Return voting weight associated with given authority or default 1 in case mapping does not exist
+	pub(crate) fn vote_weight(&self, authority: &AuthorityId) -> VoteWeight {
+		match self.voting_weights.binary_search_by(|(auth, _)| auth.cmp(authority)) {
```

Comment:
binary search only works if the vec is sorted, which it isn't - you need to sort it when building it

Reply:
during construction of `voting_weights` I am folding to `BTreeMap` which should handle sorting and deduplication for me:
```rust
			// We are sorting and deduplicating elements, which is later used by vote_weight function
			.fold(BTreeMap::new(), |mut acc, item| {
				*acc.entry(item.to_owned()).or_insert(0) += 1;
				acc
			})
```

Reply:
why not just keep `BTreeMap` instead of `Vec`? these assumptions are brittle, one might change that folding code without knowing there is a hidden assumption that the folding during build produces sorted collection.

Reply:
oh I think it evaluated and now it is not necessary, changed to BTreeMap, thanks!